### PR TITLE
Use a rails application template for Heroku + example-app

### PR DESCRIPTION
**Pull request by elia**
**Description**

The current heroku deploy button is partially broken and uses a custom buildpack. 
This PR introduces a reusable [rails application template](https://guides.rubyonrails.org/rails_application_templates.html) that can be used to generate and update a demo-app (presumably published at https://github.com/solidusio/solidus-example-app).

The app will have its own app.json and will be a good base for deploys to heroku, but also
for forks. Heroku deploys should be already on the right path for production use (cache, images, etc.)

The template can also be used as an executable to build the official demo app:

```bash
~/Code/solidus $ cd ..
# the default APP_NAME is solidus-example-app
~/Code $ solidus/bin/rails-application-template [APP_NAME]
# …generates the app…
```

**Demo app**

The resulting code is available here: https://github.com/nebulab/solidus-example-app
I just had to change the heroku button url to point at the _nebulab_ org instead of _solidusio_, other than that it's straight out of the template.

See it in action: https://elia-solidus-demo.herokuapp.com

**Screenshots**

image

image

image

**Checklist:**
- [x] I have followed [Pull Request guidelines](https://github.com/solidusio/solidus/blob/master/CONTRIBUTING.md#pull-request-guidelines)
- [x] I have added a detailed description into each commit message
- [x] I have updated Guides and README accordingly to this change (if needed)
- [x] I have added tests to cover this change (if needed)

### CODE REVIEW

File: bin/rails-application-template [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+# This file can both be used as rails application template and as executable
+if $0 == __FILE__
+  require 'shellwords'
+  app_name = ARGV.shift || default_app_name
+  options = ARGV
+  options.unshift "--database=postgresql"
+  options.unshift "--force" if app_name == default_app_name
+  command = "rails", "new", app_name, "-m", __FILE__, *options
+  warn "This file is intended to be used as Rails Application Template"
```
Comment (jacobherrington) [inline bin/rails-application-template]: "This file is intended to be used as *a* Rails Application Template"

File: bin/rails-application-template [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  rails_command "db:create"
+  rails_command "db:migrate"
+
+  git :init
+  git add: "."
+  git commit: %{ -m 'Generate app with solidus-example-app template' }
+
+  if ENV['UPDATE_EXAMPLE_APP_REPO']
+    git remote: %{ add origin git@github.com:solidusio/solidus-example-app.git }
+    git push: %{ --force --set-upstream origin master }
```
Comment (jacobherrington) [inline bin/rails-application-template]: I wasn't sure about using `--force` here without asking the user for confirmation, but I talked myself into being okay with it.

**Comment by elia** [comment]
> Looks good to me. This does make some assumptions as to services like Cloudinary and Memcachier but since this is meant as an example app I think those are fine choices that are easily changed by the user.

Yes, given it's going to be deployed on heroku I looked for services that had a free tier and were at the same time viable for production use (using a cache and a CDN are both very good practices for any production store).

File: bin/rails-application-template (hunk 2 above)
Comment (elia) [reply bin/rails-application-template]: The idea is to have something that can be automated and requires as few interaction as possible. That's only for the chore of updating the example app after a release of course, it doesn't happen when used as an app template.

Comment (jacobherrington) [reply bin/rails-application-template]: Yes, and worst case scenario we just regenerate the example app. The repo we are pushing to will not contain anything we would be afraid to lose.

**Comment by kennyadsl** [comment]
@elia I moved the repo from `nebulab` org to `solidusio`: https://github.com/solidusio/solidus-example-app

Are we good to go?

**Comment by elia** [comment]
> Are we good to go?

I'll remove the last commit and we're good! 🚀

**Comment by elia** [comment]
@kennyadsl done, and also regenerated the app to with the solidusio heroku button link 👍

**Pull request merged**